internal/app/init/main.go

```diff
@@ -48,6 +50,31 @@ func kmsg(prefix string) (*os.File, error) {
 	return out, nil
 }
 
+func container() (err error) {
+	log.Println("preparing container based deploy")
+
+	log.Println("remounting volumes as shared mounts")
+	targets := []string{"/", "/var/lib/kubelet", "/etc/cni"}
+	for _, t := range targets {
+		if err = unix.Mount("", t, "", unix.MS_SHARED, ""); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("reading the user data: %s\n", constants.UserDataPath)
+	var data *userdata.UserData
+	if data, err = userdata.Open(constants.UserDataPath); err != nil {
+		return err
+	}
+
+	log.Println("preparing the root filesystem")
+	if err = rootfs.Prepare("", true, data); err != nil {
```

Comment:
should we s/true/inContainer/ ?

Reply:
On second thought, we should be explicit here I think. Really we ought to improve the rootfs package to have a "strategy" for container based deploys and avoid this extra arg. `inContainer` is really only useful in the main func and we should always have `true` here.